subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/JavaToolchainBuildOperationsIntegrationTest.groovy

```diff
@@ -433,9 +440,15 @@ class JavaToolchainBuildOperationsIntegrationTest extends AbstractIntegrationSpe
         assertToolchainUsages(events, jdkMetadata, "JavaLauncher")
 
         where:
-        task             | _
-        ":compileKotlin" | _
-        ":test"          | _
+        kotlinPlugin | task
+        "latest"     | ":compileKotlin"
+        "1.7"        | ":compileKotlin"
+        "1.6"        | ":compileKotlin"
+        // Even though we only configure the toolchain within the `kotlin` block,
+        // it actually affects the java launcher selected by the test task.
+        "latest"     | ":test"
```

Comment:
Would it make sense to test both tasks in one go?

Reply:
Done